src/protocols/secure_channel/CheckinMessage.cpp

```diff
@@ -92,6 +86,23 @@ CHIP_ERROR CheckinMessage::ParseCheckinMessagePayload(Crypto::Aes128KeyHandle &
     return err;
 }
 
+CHIP_ERROR CheckinMessage::GenerateCheckInMessageNonce(const Crypto::Hmac128KeyHandle & hmacKeyHandle, CounterType counter,
+                                                       MutableByteSpan & output)
+{
+    VerifyOrReturnError(output.size() >= CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES, CHIP_ERROR_INVALID_ARGUMENT);
+
+    uint8_t nonceWorkBuffer[CHIP_CRYPTO_HASH_LEN_BYTES] = { 0 };
+
+    chip::Crypto::HMAC_sha shaHandler;
+    ReturnErrorOnFailure(shaHandler.HMAC_SHA256(hmacKeyHandle, (uint8_t *) (&counter), sizeof(CounterType), nonceWorkBuffer,
+                                                CHIP_CRYPTO_HASH_LEN_BYTES));
+
+    static_assert(sizeof(nonceWorkBuffer) >= CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES, "We're reading off the end of our buffer.");
+    memcpy(output.data(), nonceWorkBuffer, CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES);
```

Comment:
This is missing `output.resize(CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES)`. Please add a test that on success, the MutableByteSpan has to be == CHIP_CRYPTO_AEAD_NONCE_LENGTH_BYTES even if at first it was more space available.

Reply:
Tests will be in a follow up as disccused.